src/data/nat/basic.lean

```diff
@@ -263,6 +263,11 @@ end
 @[simp] lemma pred_one_add (n : ℕ) : pred (1 + n) = n :=
 by rw [add_comm, add_one, pred_succ]
 
+/-- Decreasing then increasing via `pred` then `succ` is idempotent
+only if the term is not 0 -/
+@[simp] lemma pred_succ_ne_zero (n : ℕ) (h : n ≠ 0) : n.pred.succ = n :=
```

Comment:
```suggestion
@[simp] lemma succ_pred_eq_of_ne_zero (n : ℕ) (h : n ≠ 0) : n.pred.succ = n :=
```

I think the names of lemmas should be what it says when not using projection notation. The similar lemma with `n > 0` instead is called `succ_pred_eq_of_pos`.